pkg/apis/gateway/v1alpha1/types.go

```diff
@@ -0,0 +1,94 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Gateway is the definition of a gateway-controller resource
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Gateway struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Status            NodePhase   `json:"type" protobuf:"bytes,2,opt,name=type"`
+	Spec              GatewaySpec `json:"spec" protobuf:"bytes,3,opt,name=spec"`
+}
+
+// GatewayList is the list of Gateway resources
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type GatewayList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Items           []Gateway `json:"items" protobuf:"bytes,2,opt,name=items"`
+}
+
+// GatewaySpec represents gateway-controller specifications
+type GatewaySpec struct {
+	// Image is the image provided by user
+	Image string `json:"image" protobuf:"bytes,1,opt,name=image"`
+
+	// ImagePullPolicy for pulling the image
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
+
+	// Command is command to run user's image
+	Command string `json:"command" protobuf:"bytes,2,opt,name=command"`
+
+	// Todo: does this needed to specified separately?
+	// ConfigMap is name of the configmap user code can access if required
+	ConfigMap string `json:"configmap,omitempty" protobuf:"bytes,3,opt,name=configmap"`
+
+	// Type is type of the gateway used as event type
+	Type string `json:"type" protobuf:"bytes,5,opt,name=type"`
+
+	// Version is used for marking event version
+	Version string `json:"version" protobuf:"bytes,6,opt,name=version"`
+
+	// Service is the name of the service to expose the gateway
+	Service Service `json:"service,omitempty" protobuf:"bytes,7,opt,name=service"`
+
+	// Sensors are list of sensors to dispatch events to
+	Sensors []string `json:"sensors" protobuf:"bytes,8,opt,name=sensors"`
+
+	// ServiceAccountName is name of service account to run the gateway
+	ServiceAccountName string `json:"serviceAccountName" protobuf:"bytes,9,opt,name=service_account_name"`
```

Comment:
the protobuf naming scheme should be camelCase: `serviceAccountName`

Reply:
yup.